### PR TITLE
Adding necessary credential changes for azure

### PR DESCRIPTION
**Pull request by deepakraj1997**

### CODE REVIEW

File: docs/examples/manifests/mssql-backup.yaml [hunk 1]
```diff
@@ -6,5 +6,5 @@ metadata:
 spec:
   includedNamespaces:
   - mssql-persistent
-  storageLocation: dpa-sample-1
+  storageLocation: d-velero-sample-1
```
Comment (weshayutin) [inline docs/examples/manifests/mssql-backup.yaml:9]: looks good so far :) qq.  why change the storage location name, is that temporary?

File: tests/e2e/common.go [hunk 2]
```diff
@@ -97,6 +92,35 @@ func decodeJson(data []byte) (map[string]interface{}, error) {
 	return jsonData, err
 }
 
+func getJsonData(path string) (map[string]interface{}, error) {
+	// Return buffer data for json
+	jsonData, _ := ioutil.ReadFile(path)
```
Comment (weshayutin) [inline tests/e2e/common.go:97]: so the old getJsonData returned jsonData, err.  Is it safe to null the error here?

File: tests/e2e/e2e_suite_test.go [hunk 3]
```diff
@@ -36,14 +37,10 @@ func init() {
 
 func TestOADPE2E(t *testing.T) {
 	flag.Parse()
-	s3Buffer, err := getJsonData(bucketFilePath)
+	s3Data, err := getJsonData(bucketFilePath)
```
Comment (weshayutin) [inline tests/e2e/e2e_suite_test.go:40]: ya.. so err here would always be null if iiuc? right?

File: tests/e2e/velero_deployment_suite_test.go [hunk 4]
```diff
@@ -36,32 +37,52 @@ var _ = Describe("Configuration testing for DPA Custom Resource", func() {
...
+						"storageAccountKeyEnvVar": vel.azureConfig.BslStorageAccountKeyEnvVar,
+						// "storageAccountKeyEnvVar": vel.azureConfig.BslStorageAccountKeyEnvVar,
+					}
+					installCase.DpaSpec.Configuration.Velero.DefaultPlugins = append(installCase.DpaSpec.Configuration.Velero.DefaultPlugins, oadpv1alpha1.DefaultPluginMicrosoftAzure)
+					installCase.DpaSpec.SnapshotLocations = []oadpv1alpha1.SnapshotLocation{
+						{
+							Velero: &velero.VolumeSnapshotLocationSpec{
+								Provider: vel.Provider,
+								Config: map[string]string{
+									"subscriptionId": vel.azureConfig.VslSubscriptionId,
```
Comment (weshayutin) [inline tests/e2e/velero_deployment_suite_test.go:85]: this is nice :)

File: tests/e2e/velero_deployment_suite_test.go [hunk 5]
```diff
@@ -597,104 +526,223 @@ var _ = Describe("Configuration testing for DPA Custom Resource", func() {
...
-				BackupLocations: []oadpv1alpha1.BackupLocation{
-					{
-						Velero: &velero.BackupStorageLocationSpec{
-							Provider: provider,
-							Default:  true,
-							StorageType: velero.StorageType{
-								ObjectStorage: &velero.ObjectStorageLocation{
-									Bucket: bucket,
-									Prefix: veleroPrefix,
+	)
```
Comment (weshayutin) [inline tests/e2e/velero_deployment_suite_test.go:529]: wow.. nice big change.. you'll have to explain to your dumb manager why it's only if provider aws and no clauses for the other providers here.

File: docs/examples/manifests/mssql-backup.yaml (hunk 1 above)
Comment (deepakraj1997) [reply docs/examples/manifests/mssql-backup.yaml:9]: I will remove this, this was just for testing!

**Review by weshayutin — COMMENTED** [review]
So the e2e that passed here ran against GCP afaict.
-credentials=/var/run/oadp-credentials/gcp-credentials \
-velero_bucket=/var/run/oadp-credentials/gcp-velero-bucket-name \
-velero_namespace=openshift-adp \
-creds_secret_ref=cloud-credentials-gcp \
-velero_instance_name=velero-sample \
-bsl_region=us-east-1 \
-vsl_region=us-central1 \
-bsl_profile=migration-engineering \
-provider=gcp \
-timeout_multiplier=1 \
-openshift_ci=true \
-ci_cred_file=/var/run/secrets/ci.openshift.io/cluster-profile/gce.json


Do we need to exercise the change against ec2 and azure? prior to merge?

**Comment by deepakraj1997** [comment]
Yes, we need to change run this against gcp and azure. For that I will have to make some changes in the ci private branch config.

**Comment by deepakraj1997** [comment]
I am merging this to the branch to test the azure changes.

**Pull request merged**